### PR TITLE
- Port to Python 3

### PR DESCRIPTION
**Pull request by rjschwei**
- Some unrelated code cleanliness

### CODE REVIEW

File: lib/serviceAccessConfig/generatorfactory.py [hunk 1]
```diff
@@ -64,16 +64,22 @@ def get_access_rule_generators(access_generator_config):
                     globals(), locals(),
                     fromlist=[class_name]
                 )
-                exec 'generator = module.%s("%s")' % (
-                    class_name,
-                    ip_source_config_file_name
+                exec(
```
Comment (smarlowucf) [inline lib/serviceAccessConfig/generatorfactory.py]: Not sure exactly what this is doing but could probably use getattr in some way instead of exec.

File: lib/serviceAccessConfig/generatorfactory.py [hunk 2]
```diff
@@ -64,16 +64,22 @@ def get_access_rule_generators(access_generator_config):
...
+                    'generator = module.%s("%s")' % (
+                        class_name,
+                        ip_source_config_file_name
+                    ),
+                    globals(),
+                    locals()
                 )
                 try:
-                    generator.set_config_values(access_generator_config)
+                    locals()['generator'].set_config_values(
```
Comment (smarlowucf) [inline lib/serviceAccessConfig/generatorfactory.py]: Why is this needed? If pulling `generator` from locals it should be available directly as `generator`.

File: lib/serviceAccessConfig/haproxy.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#  Copyright (C) 2016 SUSE LLC, Robert Schweikert <rjschwei@suse.com>
+#  Copyright (C) 2019 SUSE LLC, Robert Schweikert <rjschwei@suse.com>
```
Comment (smarlowucf) [inline lib/serviceAccessConfig/haproxy.py:1]: Line 83 undefined name 'counter'

**Review by smarlowucf — CHANGES_REQUESTED** [review]
Consider running flake8 against the branch as there's many warnings in the test modules. `setup.py` also has a syntax error at https://github.com/SUSE-Enceladus/service-access-config/blob/master/setup.py#L31.

File: lib/serviceAccessConfig/haproxy.py (hunk 3 above)
Comment (smarlowucf) [reply lib/serviceAccessConfig/haproxy.py:1]: https://github.com/SUSE-Enceladus/service-access-config/blob/1335fdefa250a0af5fd2f750182cb616307d8f0e/lib/serviceAccessConfig/haproxy.py#L83

File: lib/serviceAccessConfig/generatorfactory.py (hunk 2 above)
Comment (rjschwei) [reply lib/serviceAccessConfig/generatorfactory.py]: That's what one would think but at least for me the tests failed on Python 3.4 with "generator" not being available after exec.

I tried pretty much everything I could think of to accommodate exec in Py 3, butthis was the only thing that ended up working.

I do agree with you, but it didn't work that way.

File: Makefile [hunk 4]
```diff
@@ -26,7 +26,10 @@ install:
 
 pep8:
 	@python -m pycodestyle lib/serviceAccessConfig/*.py
-	@python -m pycodestyle --ignore=E402 tests/unit/*.py 
+	@python -m pycodestyle --ignore=E402 tests/unit/*.py
```
Comment (smarlowucf) [inline Makefile]: flake8 runs both pycodestyle and pyflakes. So calls direct to pycodestyle can be dropped as they are duplicative.

File: lib/serviceAccessConfig/generatorfactory.py (hunk 2 above)
Comment (smarlowucf) [reply lib/serviceAccessConfig/generatorfactory.py]: Strange, I don't see the behavior in 3.7. The results in exec are added to locals. But can't this still be simplified from:

```
exec(
    'generator = module.%s("%s")' % (
        class_name,
        ip_source_config_file_name
    ),
    globals(),
    locals()
)
```

to:

```
generator = getattr(module, class_name)(ip_source_config_file_name)
```

Which solves the problem?

**Review by smarlowucf — APPROVED** [review]
